Update Lambda Python runtime and IAM policy

mbc3k opened the pull request:
python2.7 is no longer supported so new installations cannot use it. Updated to 3.9.

Also replaced "AWSLambdaFullAccess" managed policy because it is deprecated. "AWSLambdaBasicExecutionRole" is now also required because it allows Lambdas to log to Cloudwatch Logs.

mbc3k committed:
Update Lambda managed policy

AWSLambdaFullAccess is deprecated (and doesn't exist on newly created accounts) so this change replaces it with AWSLambda_FullAccess.

Also, add AWSLambdaBasicExecutionRole managed service role so that lambdas
can write to Cloudwatch Logs.

mbc3k committed:
Update Lambda runtime from 2.7 to 3.9

This is the runtime that AWS recommends as of right now. The Python code
does not seem to require any updates in order to run.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/puppetlabs/aws_resource_reaper?pullRequest=21) <br/>All committers have signed the CLA.

mbc3k committed:
Run 2to3 on slack_notifier.py, update urllib syntax

urllib2 doesn't exist anymore so use urllib instead.

mbc3k committed:
Decode byte string in UTF-8 format to standard Python string

The ast module was reporting "ValueError: malformed node or string" while
validating.

mbc3k committed:
Craft the HTTP response as bytes the Python 3 way

Without these changes, the Response package causes a crash because it receives
a string rather than bytes. We craft the response and convert it to bytes, then
send that to Response.

mbc3k committed:
Remove unused variable, join was causing TypeErrors

It looks like the point of this was to convert the object that Request
returns into a string. Since we don't use the variable that holds this string,
let's just delete it.

mbc3k committed:
Convert the post mechanism from deprecated incoming webhook to workflow

Raw incoming webhooks have been deprecated by Slack in favor of apps. Apps
often require approval from Slack administrators, whereas Workflows do not
require approval since they are there to begin with.

mbc3k committed:
Update the README to mention a workflow rather than an app

mbc3k commented:
I also had to adjust the Slack notifier so it uses Slack's new "Workflow" rather than using an app. We lose the ability to fine-tune the formatting of the notifications, but we gain ease of use. Apps have to be approved by Slack admins whereas Workflows are built in. Their JSON format is much simpler as well.

mbc3k committed:
Fix README.md formatting
